test/Libraries/RevitNodesTests/Elements/ElementTests.cs

```diff
@@ -260,31 +261,84 @@ public void ElementCurveReferences_ExtractsExpectedReferences()
+            Assert.AreEqual(4, hostedElementsIncludeEmbeddedWallsAndEmbeddedInserts.Count);
+            Assert.AreEqual(5, hostedElementsIncludeEmbeddedWallsAndEmbeddedInsertsAndOpenings.Count);
+
+            Assert.AreEqual(3, hostedElementsIncludeEmbeddedInserts.Count);
+            Assert.AreEqual(4, hostedElementsIncludeSEmbeddedInsertsAndOpenings.Count);
+            Assert.AreEqual(4, hostedElementsIncludeEmbeddedInsertsAndOpeningsAndShadows.Count);
+
+            //Assert all combinations has the right elements as output
+            AssertExpectedElementNameCollection(new[] { "600 x 3100", "600 x 3100", "600 x 3100" }, hostedElementsIncludeNothing);
+            AssertExpectedElementNameCollection(new[] { "Curtain Wall", "600 x 3100", "600 x 3100", "600 x 3100", "Rectangular Straight Wall Opening" }, hostedElementsIncludeEverything);
+
+            AssertExpectedElementNameCollection(new[] { "600 x 3100", "600 x 3100", "600 x 3100", "Rectangular Straight Wall Opening" }, hostedElementsIncludeOpenings);
+            AssertExpectedElementNameCollection(new[] { "600 x 3100", "600 x 3100", "600 x 3100", "Rectangular Straight Wall Opening" }, hostedElementsIncludeOpeningsAndShadows);
+            AssertExpectedElementNameCollection(new[] { "Curtain Wall", "600 x 3100", "600 x 3100", "600 x 3100", "Rectangular Straight Wall Opening" }, hostedElementsIncludeOpeningsAndShadowsAndEmbeddedWalls);
+
+            AssertExpectedElementNameCollection(new[] { "600 x 3100", "600 x 3100", "600 x 3100"}, hostedElementsIncludeShadows);
+            AssertExpectedElementNameCollection(new[] { "Curtain Wall", "600 x 3100", "600 x 3100", "600 x 3100" }, hostedElementsIncludeShadowsAndEmbeddedWalls);
+            AssertExpectedElementNameCollection(new[] { "Curtain Wall", "600 x 3100", "600 x 3100", "600 x 3100" }, hostedElementsIncludeShadowsAndEmbeddedWallsAndEmbeddedInserts);
+
+            AssertExpectedElementNameCollection(new[] { "Curtain Wall", "600 x 3100", "600 x 3100", "600 x 3100" }, hostedElementsIncludeEmbeddedWalls);
+            AssertExpectedElementNameCollection(new[] { "Curtain Wall", "600 x 3100", "600 x 3100", "600 x 3100" }, hostedElementsIncludeEmbeddedWallsAndEmbeddedInserts);
+            AssertExpectedElementNameCollection(new[] { "Curtain Wall", "600 x 3100", "600 x 3100", "600 x 3100", "Rectangular Straight Wall Opening" }, hostedElementsIncludeEmbeddedWallsAndEmbeddedInsertsAndOpenings);
+
+            AssertExpectedElementNameCollection(new[] { "600 x 3100", "600 x 3100", "600 x 3100" }, hostedElementsIncludeEmbeddedInserts);
+            AssertExpectedElementNameCollection(new[] { "600 x 3100", "600 x 3100", "600 x 3100", "Rectangular Straight Wall Opening" }, hostedElementsIncludeSEmbeddedInsertsAndOpenings);
+            AssertExpectedElementNameCollection(new[] { "600 x 3100", "600 x 3100", "600 x 3100", "Rectangular Straight Wall Opening" }, hostedElementsIncludeEmbeddedInsertsAndOpeningsAndShadows);
```

Comment:
we're repeating ourselves a lot here with these arrays, in the `Arrange` section we should either
- declare a var for each "600x".. and use those so we don't duplicate error-prone strings
- above plus declare a few lists that hold the expected outcomes